motor/providers/aws/provider.go

```diff
@@ -88,8 +88,13 @@ func New(pCfg *providers.Config, opts ...ProviderOption) (*Provider, error) {
 	// gather information about the aws account
 	identity, err := CheckIam(t.config)
```

Comment:
This is a great addition. I wonder if we could simplify the implementation for more clarity. This can be done in a followup PR. Something like:

```go
regions := []string{}

if cfg.Region != "" {
   regions = []string{cfg.Region}
} else {
   regions  = []string{"us-east-1", "us-gov-west-1"} // all regions we try before giving up
}
 
for _, region range regions {
   identity, err := CheckIam(t.config)
   if err != nil { continue}
   
   // all passed
   found = true
   break
}

if found != true {
  // do what is required if we cannot connect
}

// proceed
```


```

Reply:
oh yes, that's nice!